libs/build-nss-desktop.sh

```diff
@@ -98,12 +105,18 @@ cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libpkcs7.a" "${DIST_DIR}/lib"
 cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libsmime.a" "${DIST_DIR}/lib"
 cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libsoftokn_static.a" "${DIST_DIR}/lib"
 cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libssl.a" "${DIST_DIR}/lib"
-cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libhw-acc-crypto-avx.a" "${DIST_DIR}/lib"
-cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libhw-acc-crypto-avx2.a" "${DIST_DIR}/lib"
 
-# HW specific.
-# https://searchfox.org/mozilla-central/rev/1eb05019f47069172ba81a6c108a584a409a24ea/security/nss/lib/freebl/freebl.gyp#159-163
-cp -p -L "${NSS_DIST_OBJ_DIR}/lib/libgcm-aes-x86_c_lib.a" "${DIST_DIR}/lib"
+#Apple M1 need HW specific libs copied over to successfully build
+if [[ "${TARGET_OS}" == "macos" ]] && [[ "${TARGET_ARCH}" == "aarch64" ]]; then
```

Comment:
I think this is a good suggestion! The only issue is Linux is also used heavily in cross-compilation and from what I understand, uname is a little tricky with that kind of stuff, so this kinda errs on the side of caution to prevent any unwanted side-effects. Could be wrong though!

Reply:
I think Ben's correct - I don't think the cross compilation stuff would break if you did this. I do recall thinking we had a good reason for that other check yesterday but now I can't see it :)

(I think that ultimately we will want the cross compilation cruft to target this processor, which IIUC doesn't happen, but that's something we can add later if we need it (tbh, I'm not sure how these cross-compilation targets are used in practice)

Reply:
Since we only set target_arch in the above macos statement, this will be safe to remove though. Thanks Ben for the suggestion! Will implement shortly.